cmd/sdk-server/main.go

```diff
@@ -77,7 +78,11 @@ func main() {
 		time.Sleep(time.Duration(ctlConf.Delay) * time.Second)
 	}
 
-	ctx := signals.NewSigKillContext()
+	ctx := context.Background()
+	if !runtime.FeatureEnabled(runtime.FeatureGracefulTerminationFilter) {
```

Comment:
I'm not sure I understand why this conditional is here. 

* If the feature is enabled and the sidecar is running is test or local mode this changes the behavior to use the background context instead of the sigkillcontext. 
* If the feature is enabled, the context gets replaced below with the newsdkservercontext. That function can be passed `context.Background()` instead of `ctx` to have the same effect. 

Do we want to change the context that is used in the test and local configurations? Is there a reason for them to use the sigkillcontext or the sdkserver context? I'm wondering why the new code path is only installed when running as a sidecar below and not always, e.g.

```
ctx := signals.NewSigKillContext()
if runtime.FeatureEnabled(runtime.FeatureGracefulTerminationFilter) {
  ctx = s.NewSDKServerContext(context.Background())
}
```

Reply:
thanks for the suggestion! :) updated the conditional